src/components/tasks/card/progressContainer/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { FC, useState } from 'react';
+import { useRouter } from 'next/router';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+import {
+    useUpdateSelfTaskMutation,
+    useUpdateTaskMutation,
+} from '@/app/services/tasksApi';
+import useUserData from '@/hooks/useUserData';
+import { useGetUserQuery } from '@/app/services/userApi';
+import ProgressText from './ProgressText';
+import Progressbar from './ProgressBar';
+import { ERROR_MESSAGE, PROGRESS_SUCCESSFUL } from '@/constants/constants';
+import classNames from '@/components/tasks/card/card.module.scss';
+import { ProgressContainerProps } from '@/interfaces/task.type';
+
+const ProgressContainer: FC<ProgressContainerProps> = ({ content }) => {
+    const router = useRouter();
+    const { dev } = router.query;
+
+    const [isProgressMade, setIsProgressMade] = useState<boolean>(false);
+    const [progressValue, setProgressValue] = useState<number>(0);
+
+    const { isUserAuthorized } = useUserData();
+    const { data: userData } = useGetUserQuery();
+    const [updateTask, { isLoading: isLoadingUpdateTaskDetails }] =
+        useUpdateTaskMutation();
+    const [updateSelfTask, { isLoading: isLoadingSelfTaskUpdate }] =
+        useUpdateSelfTaskMutation();
+    const checkingLoading =
```

Comment:
Can be fixed in later PRs. The name doesn't imply that this is a boolean. Better keep it `isLoading` or something similar